types/src/system/auction/bid/vesting.rs

```diff
@@ -74,23 +67,31 @@ impl VestingSchedule {
         let locked_amounts_length =
             vesting_schedule_period_to_weeks(vesting_schedule_period_millis);
 
+        if locked_amounts_length as usize >= LOCKED_AMOUNTS_MAX_LENGTH {
+            // We will not initialize a vesting schedule if the genesis configuration is incorrect.
+            // There are restrictions put in place in the chainspec config to make sure we will not
+            // hit this case.
+            //
+            // This is done to avoid breaking changes in the serialization format.
+            return false;
+        }
```

Comment:
I think we should panic here, since returning `false` doesn't really convey the error.
```suggestion
        assert!(
            locked_amounts_length < LOCKED_AMOUNTS_LENGTH as u64,
            "vesting schedule period must be less than {} weeks",
            LOCKED_AMOUNTS_LENGTH
        );
```

Let's also add the following to the doc comment:
```rust
    ///
    /// # Panics
    ///
    /// Panics if `vesting_schedule_period_millis` represents more than 13 weeks.
```

Reply:
Addressed